sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -57,7 +57,8 @@ case class BroadcastHashJoinExec(
   }
 
   override lazy val metrics = Map(
-    "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
+    "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"),
+    "numMatchedPairs" -> SQLMetrics.createMetric(sparkContext, "number of matched pairs"))
```

Comment:
`numMatchedPairs` -> `numJoinedRows`?

Reply:
Sure, do you also want me to change the display name as well, `number of matched pairs` -> `number of joined rows`? I feel that `number of matched pairs` might be more precise and less confusing to the user. But if you think that  `number of joined rows` better I will change the name of the metric accordingly.

Reply:
>  I feel that number of matched pairs might be more precise and less confusing to the user.

Any reason? At first, I was not sure about what `pairs` means though.

Reply:
How about `number of matched rows` instead?  I wanted highlight the fact that this metric measures the number of keys matched between the stream side and the build side.  I also felt that users might not recognize the difference between `number of joined rows` and `number of output rows`. But `number of joined rows` is fine as well. If you think that users will understand that better I will change the name of the metric to `number of joined rows`

Reply:
sgtm

Reply:
Updated the name of the metric to `number of matched rows` in the latest commit.